### PR TITLE
85223 create fss api

### PR DESCRIPTION
**Pull request by HussenShambeshDfE**
Implemented new V2 endpoint to get all FSS Projects.

https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_workitems/edit/85223

### CODE REVIEW

File: TramsDataApi/Controllers/V2/FSSProjectController.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+{
+    [ApiVersion("2.0")]
+    [ApiController]
+    [Route("v{version:apiVersion}/fss/")]
+    public class FssProjectController : Controller
+    {
+        private readonly ILogger<FssProjectController> _logger;
+        private readonly IGetAllFssProject _getAllFssProject;
+
+        public FssProjectController(ILogger<FssProjectController> logger, IGetAllFssProject getAllFssProject)
```
Comment (kolevk) [inline TramsDataApi/Controllers/V2/FSSProjectController.cs:18]: should this` IGetAllFssProject getAllFssProject` be `IGetAllFssProjects fssProjects`

File: TramsDataApi/UseCases/IGetAllFssProject.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+using TramsDataApi.ResponseModels;
+
+namespace TramsDataApi.UseCases
+{
+    public interface IGetAllFssProject
```
Comment (kolevk) [inline TramsDataApi/UseCases/IGetAllFssProject.cs:6]: IGetAllFssProjects?

File: TramsDataApi/UseCases/GetAllFssProject.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using TramsDataApi.Factories;
+using TramsDataApi.Gateways;
+using TramsDataApi.ResponseModels;
+
+namespace TramsDataApi.UseCases
+{
+    public class GetAllFssProject : IGetAllFssProject
```
Comment (kolevk) [inline TramsDataApi/UseCases/GetAllFssProject.cs:9]: Can this be `GetAllFssProjects : IGetAllFssProjects`?

Comment (HussenShambeshDfE) [reply TramsDataApi/UseCases/GetAllFssProject.cs:9]: Yes can be

File: TramsDataApi/Controllers/V2/FSSProjectController.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            _logger.LogInformation($"Retreiving all FSS Projects ");
+
+            var projects = _getAllFssProjects.Execute().ToList();
+
+            _logger.LogInformation($"Found {0} projects, " , projects.Count);
+
+            _logger.LogDebug(JsonSerializer.Serialize(projects));
+
+            // Since the dat sent s not being paginated passing null in the pagingResponse parameter
```
Comment (MartinWheelerMT) [inline TramsDataApi/Controllers/V2/FSSProjectController.cs:36]: can we remove this comment, or correct the errors in it (missing 'a' from data, missing 'i' from is) please

**Review by MartinWheelerMT — APPROVED** [review]

**Review by kolevk — APPROVED** [review]

**Pull request merged**